js/core/src/Blocks.svelte

```diff
@@ -227,13 +228,23 @@
 		return get_data(component_id);
 	}
 
+	function check_inputs_ready(): boolean {
+		if (inputs_waiting.length > 0) {
+			return false;
+		}
+		return true;
+	}
+
 	async function trigger_api_call(
 		dep_index: number,
 		trigger_id: number | null = null,
 		event_data: unknown = null
 	): Promise<void> {
 		let dep = dependencies.find((dep) => dep.id === dep_index)!;
-
+		if (check_inputs_ready() === false) {
```

Comment:
I haven't checked but I think we need to take the dep_index into account in this case. Otherwise a pending upload for event B can block event A.

Reply:
Ah yes, fixed!